### PR TITLE
Support X-Debug-Id and X-SF-Tracing-Debug in httpsink

### PR DESCRIPTION
**Pull request by mdubbyap**

### CODE REVIEW

File: sfxclient/httpsink.go [hunk 1]
```diff
@@ -126,22 +129,42 @@ func (h *HTTPSink) doBottom(ctx context.Context, f func() (io.Reader, bool, erro
...
 	req.Header.Set(TokenHeaderName, h.AuthToken)
 	req.Header.Set("User-Agent", h.UserAgent)
 	req.Header.Set("Connection", "keep-alive")
+	if v := ctx.Value(xDebugID); v != nil {
+		req.Header.Set(string(xDebugID), v.(string))
+	}
+	if v := ctx.Value(xTracingDebug); v != nil {
+		hn, _ := os.Hostname()
+		req.Header.Set(string(xTracingDebug), v.(string))
+		req.Header.Set(xTracingID, hn+"-"+strconv.FormatInt(rand.Int63(), 10))
```
Comment (jgheewala) [inline sfxclient/httpsink.go]: how often we will be enabling Tracing-Debug?

File: sfxclient/httpsink.go [hunk 2]
```diff
@@ -126,22 +129,44 @@ func (h *HTTPSink) doBottom(ctx context.Context, f func() (io.Reader, bool, erro
...
+}
+
+type xKeyContextValue string
+
+var (
+	// XDebugID Debugs the transaction via signalscope if value matches known secret
+	XDebugID xKeyContextValue = "X-Debug-Id"
+	// XTracingDebug Sets debug flag on trace if value matches known secret
+	XTracingDebug xKeyContextValue = "X-SF-Trace-Token"
+	xTracingID                     = "X-SF-Tracing-ID"
```
Comment (mpetazzoni) [inline sfxclient/httpsink.go]: What does this header control?

File: sfxclient/httpsink.go (hunk 1 above)
Comment (mdubbyap) [reply sfxclient/httpsink.go]: Very rarely, why?

File: sfxclient/httpsink.go (hunk 2 above)
Comment (mdubbyap) [reply sfxclient/httpsink.go]: Updated with a comment.

File: sfxclient/httpsink.go (hunk 1 above)
Comment (jgheewala) [reply sfxclient/httpsink.go]: was just wondering how many time `+` will be done.

**Pull request merged**